hw-model/src/lib.rs

```diff
@@ -395,6 +406,11 @@ pub trait HwModel {
         }
     }
 
+    /// Toggle reset pins and wait for ready_for_fuses
+    fn warm_reset(&mut self) {
+        panic!("warm_reset unimplemented");
```

Comment:
Can we add a tracking Github issue to implement this on the sw emulator?

One thing that concerns me is we may not catch any regressions to code that interacts with warm reset, since the test automation only runs against sw-emulator.